extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+class StaffWelcomePoster {
+
+	const LANG_TO_STAFF_MAP = [
+		'de' => 24961680, // ForestFairy
+		'en' => 26339491, // Mira Laime
+		'es' => 12648798, // Luchofigo85
+		'fr' => 26442523, // Hypsoline
+		'it' => 3279487,  // Leviathan_89
+		'ja' => 29395778, // Kuro0222
+		'ko' => 24883131, // Miri-Nae
+		'nl' => 4142476,  // Yatalu
+		'pl' => 1117661,  // Nanaki
+		'pt' => 24005296, // Macherie ana
+		'ru' => 26457441, // Vlazovskiy
+		'vi' => 26041741, // KhangND
+		'zh' => 11909873  // Cal-Boy
+	];
+
+	const MESSAGE_KEY = 'discussions-staff-welcome-post';
+
+	const DEFAULT_LANG = 'en';
+
+	private $threadCreator;
+
+	public function __construct() {
+		$this->threadCreator = new ThreadCreator();
+	}
+
+	public function postMessage( int $siteId, string $language ): bool {
+		$trimmedLang = $this->trimLang( $language );
+		$staffId = $this->getStaffFromLang( $trimmedLang );
+		$message = $this->getMessage( $trimmedLang );
+
+		$success = $this->threadCreator->create( $staffId, $siteId,  $message );
+
+		return $success;
+	}
+
+	private function getStaffFromLang( string $language ): int {
+		$staffId = self::LANG_TO_STAFF_MAP[$language];
```

Comment:
This will throw an error for any language that doesn't exist in our core language list.  It looks like the `??` operator protects from that, so this would work:
```php
return self::LANG_TO_STAFF_MAP[$language] ?? self::LANG_TO_STAFF_MAP[self::DEFAULT_LANG];
```